docs/src/modules/utils/useLatest.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+/**
+ * Returns the last defined value that has been passed in [value]
+ */
+export default function useLatest<T>(value: T | undefined): T | undefined {
+  const latest = React.useRef(value);
+  if (value !== undefined && value !== null) {
+    latest.current = value;
+  }
+  return latest.current;
+}
```

Comment:
Not sure about having shared utils for this one yet (used once). The logic doesn't seem concurrent mode compatible 😁 (updating a ref in the render)

Reply:
I tend to use this hook fairly regularly but I don't mind bringing it back into the `SearchIcons` file until I want to reuse it. I also extracted it so I could leverage typescript.

Regarding concurrent mode compatibility, it's true that it updates a ref in a render function, but it still is side effect free. Calling this function multiple times won't result in different output. So I'd say it is concurrent mode compatible. We could update it in an effect, but the end result would be the same.